### ISSUE TITLE
Can TFLearn interface with scikit-learn's cross validation pipeline?

### ISSUE THREAD

**Issue by bwllc** [tflearn/tflearn#965]
Hi there!

I'm an experienced scikit-learn user, and I played with shallow neural nets many years ago.  I'm fairly new to deep neural nets.  I've been pretty happy with TFLearn so far.  I subclassed tflearn.DNN to make some basic network topologies I wanted to try.  I needed a custom loss function, and I wanted a fit method with callbacks that could tell when it was time to stop training, rather than grind on for many unnecessary epochs.  I've got all that working.

At this point, I want to start exploring hyperparameters: larger network topologies, batch sizes, etc.  If I was using ordinary scikit-learn models, I would import train_test_split() and cross_validate() to do this job.  I am obtaining various errors when I try this with a tflearn.DNN.  Reading through my error messages and the scikit-learn documents, my understanding is that a scikit-learn compatible Estimator expects to have certain methods, such as score() and get_params(), as hooks for cross-validation.  The tflearn.DNN class does not have these methods.  

I may be able to implement these and other methods as I understand the code better.  For now, I would like to understand why such methods do not exist in the existing TFLearn code.  Is it just because someone is needed to do the work?  I've done some additional reading that implies that most people don't optimize hyperparameters when working with DNN, because their models and data sets are simply too large.  Well, I'm not quite to that stage yet.  My data sets and models are small enough to fit in the RAM of a single computer.  Perhaps most people muddle through without using hyperparameter optimization, but this is not my plan.

Alternately, are the Estimator methods missing because there is some special obstacle imposed by Tensorflow's computational graphs?  For example, I'm having trouble understanding how one would switch dropout layers on and off at will.

Please help me to understand the current state of TFLearn.  Thanks!
